src/main/webapp/assets/js/navbar.js

```diff
@@ -1,12 +1,58 @@
-document.write('\
-<div id="nav-container">\
-    <a href="index.html" class="nav-link">Home</a>\
-    <a href="about.html" class="nav-link">About us</a>\
-    <div id="search-container">\
-        <form action="/placeholder.js">\
-            <input type="text" placeholder="Search listing...">\
-            <button type="submit"><i class="fa fa-search"></i></button>\
-        </form>\
-    </div>\
-</div>'
-);
+document.addEventListener("DOMContentLoaded", function(){ 
+  isLoggedin();
+});
+
+/**
+ * Fetch data from /login endpoint to check the status of the user.
+ * Data that the enpoint contains is JSON.
+ * If the user is logged in, the userEmail will be a non-empty string.
+ * Depending on the log in status, the navbar rendered will look different.
+ */
+function isLoggedin() {
+    console.log("trigger")
+    fetch('/login').then(response => response.json()).then(userJson => {
```

Comment:
We decided to stick to async await rather than .then() for Promises!

Reply:
my bad! i forgot!